public/app/features/explore/utils/decorators.ts

```diff
@@ -58,6 +59,11 @@ export const decorateWithFrameTypeMetadata = (data: PanelData): ExplorePanelData
       case 'flamegraph':
         flameGraphFrames.push(frame);
         break;
+      case 'plugin':
```

Comment:
Looks like this potentially could be part of the default case and if set, try using the plugin id and load the plugin. Then preferredVisualization type could support any plugin id, rather than introducing a type=plugin in the mix? Guess it might complicate things?

Reply:
I did do that originally, then thought it would be better to have a explicit type=plugin to prevent any chance of existing plugins causing issues. In the very small chance that some plugin exists with `meta.custom.pluginID`.